pallets/capacity/src/tests.rs

```diff
@@ -770,3 +772,196 @@ fn calculate_capacity_reduction_determines_the_correct_capacity_reduction_amount
+	new_test_ext().execute_with(|| {
+		let msa_id = 1;
+		assert_eq!(Capacity::balance(msa_id), BalanceOf::<Test>::zero());
+	});
+}
+
+#[test]
+fn impl_deduct_is_successful() {
+	new_test_ext().execute_with(|| {
+		let target_msa_id = 1;
+		let remaining_amount = BalanceOf::<Test>::from(10u32);
+		let total_available_amount = BalanceOf::<Test>::from(10u32);
+		let _ = create_capacity_account_and_fund(
+			target_msa_id,
+			remaining_amount,
+			total_available_amount,
+		);
+
+		assert_ok!(Capacity::deduct(target_msa_id, 5u32.into()));
+
+		let mut capacity_details = CapacityDetails::<
+			BalanceOf<Test>,
+			<Test as frame_system::Config>::BlockNumber,
+		>::default();
+
+		capacity_details.remaining = 5u32.into();
+		capacity_details.total_tokens_staked = 10u32.into();
+		capacity_details.total_available = 10u32.into();
+		capacity_details.last_replenished_epoch = 1u32.into();
+
+		assert_eq!(Capacity::get_capacity_for(target_msa_id).unwrap(), capacity_details);
+	});
+}
+
+#[test]
+fn impl_deduct_errors_target_capacity_not_found() {
+	new_test_ext().execute_with(|| {
+		let target_msa_id = 1;
+		let amount = BalanceOf::<Test>::from(10u32);
+		assert_noop!(
+			Capacity::deduct(target_msa_id, amount),
+			Error::<Test>::TargetCapacityNotFound
+		);
+	});
+}
+
+#[test]
+fn impl_deduct_errors_insufficien_balance() {
+	new_test_ext().execute_with(|| {
+		let target_msa_id = 1;
+		let remaining_amount = BalanceOf::<Test>::from(10u32);
+		let total_available_amount = BalanceOf::<Test>::from(10u32);
+		let _ = create_capacity_account_and_fund(
+			target_msa_id,
+			remaining_amount,
+			total_available_amount,
+		);
+
+		assert_noop!(
+			Capacity::deduct(target_msa_id, 20u32.into()),
+			Error::<Test>::InsufficientBalance
+		);
+
+		let mut capacity_details = CapacityDetails::<
+			BalanceOf<Test>,
+			<Test as frame_system::Config>::BlockNumber,
+		>::default();
+
+		capacity_details.remaining = 10u32.into();
+		capacity_details.total_tokens_staked = 10u32.into();
+		capacity_details.total_available = 10u32.into();
+		capacity_details.last_replenished_epoch = 1u32.into();
+
+		assert_eq!(Capacity::get_capacity_for(target_msa_id).unwrap(), capacity_details);
+	});
+}
+#[test]
+fn impl_deposit_is_successful() {
+	new_test_ext().execute_with(|| {
+		let target_msa_id = 1;
+		let remaining_amount = BalanceOf::<Test>::from(5u32);
+		let total_available_amount = BalanceOf::<Test>::from(10u32);
+		let _ = create_capacity_account_and_fund(
+			target_msa_id,
+			remaining_amount,
+			total_available_amount,
+		);
+
+		assert_ok!(Capacity::deposit(target_msa_id, 5u32.into()),);
+	});
+}
+
+#[test]
+fn impl_deposit_errors_target_capacity_not_found() {
+	new_test_ext().execute_with(|| {
+		let target_msa_id = 1;
+		let amount = BalanceOf::<Test>::from(10u32);
+		assert_noop!(
+			Capacity::deposit(target_msa_id, amount),
+			Error::<Test>::TargetCapacityNotFound
+		);
+	});
+}
+
+#[test]
+fn impl_deposit_errors_increase_exceeds_available() {
```

Comment:
It's a lot more helpful to describe the expected behavior rather than the thing that's being tested, e.g. "deposit_results_IncreaseExceedsAvailable_when_trying_to_add_too_much" or something

Reply:
I like this too. I did as you suggested for a better description. I could not do the camel case because the lint will scream. I do like how it stands out and I think we should consider updating the linting.